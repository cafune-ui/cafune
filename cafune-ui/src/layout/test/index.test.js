
  import  Layout  from '../';
  import { render } from 'enzyme';
  import { shallow, deep } from 'preact-render-spy';
    

  describe('Layout', () => {
    it('should render properly', () => {
      const wrapper = shallow(<Layout />);
      expect(wrapper).toMatchSnapshot();
    });
  });