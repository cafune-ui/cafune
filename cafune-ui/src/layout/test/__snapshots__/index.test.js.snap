// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "col": "3",
        },
        "children": Array [
          "col: 3",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "col": "7",
          "offset": "2",
        },
        "children": Array [
          "col: 7, offset: 3",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "col": "8",
        },
        "children": Array [
          "col: 8",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Layout should render properly 2`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "gutter": "20",
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "tag": "li",
        },
        "children": Array [
          "col: 8, gutter: 20",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "tag": "li",
        },
        "children": Array [
          "col: 8, gutter: 20",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "tag": "li",
        },
        "children": Array [
          "col: 8, gutter: 20",
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Layout should render properly 3`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "direction": "row",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "4",
            },
            "children": Array [
              "col: 4",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "5",
            },
            "children": Array [
              "col: 5",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "direction": "row-reverse",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "4",
            },
            "children": Array [
              "col: 4",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "5",
            },
            "children": Array [
              "col: 5",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "4",
            },
            "children": Array [
              "col: 4",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "5",
            },
            "children": Array [
              "col: 5",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "direction": "column-reverse",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "4",
            },
            "children": Array [
              "col: 4",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "5",
            },
            "children": Array [
              "col: 5",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;

exports[`Layout should render properly 4`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "wrap": "nowrap",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "wrap": "wrap",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "wrap": "wrap-reverse",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;

exports[`Layout should render properly 5`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "justify": "start",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "justify": "end",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "justify": "center",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "justify": "between",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "justify": "around",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "3",
            },
            "children": Array [
              "col: 3",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;

exports[`Layout should render properly 6`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "align": "start",
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "15",
            },
            "children": Array [
              "col: 15",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "align": "center",
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "15",
            },
            "children": Array [
              "col: 15",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "align": "end",
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "15",
            },
            "children": Array [
              "col: 15",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "align": "baseline",
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "15",
            },
            "children": Array [
              "col: 15",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "align": "stretch",
          "direction": "column",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "col": "15",
            },
            "children": Array [
              "col: 15",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "10",
            },
            "children": Array [
              "col: 10",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
          VNode {
            "attributes": Object {
              "col": "8",
            },
            "children": Array [
              "col: 8",
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;

exports[`Layout should render properly 7`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "start",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "center",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "end",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "between",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "around",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
      VNode {
        "attributes": Object {
          "class": "caf-demo-flexblock",
        },
        "children": Array [
          VNode {
            "attributes": Object {
              "alignContent": "stretch",
            },
            "children": Array [
              VNode {
                "attributes": Object {
                  "col": "15",
                },
                "children": Array [
                  "col: 15",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "10",
                },
                "children": Array [
                  "col: 10",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
              VNode {
                "attributes": Object {
                  "col": "8",
                },
                "children": Array [
                  "col: 8",
                ],
                "key": undefined,
                "nodeName": [Function],
              },
            ],
            "key": undefined,
            "nodeName": [Function],
          },
        ],
        "key": undefined,
        "nodeName": "div",
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;
